velox/exec/VectorHasher.h

```diff
@@ -278,6 +284,21 @@ class VectorHasher {
     return size == 0 ? word : word + (1L << (size * 8));
   }
 
+  template <typename T>
+  inline int64_t hashData(T value) const {
+    return value;
+  }
+
+  template <>
+  inline int64_t hashData(Date value) const {
+    return value.days();
+  }
+
+  template <>
+  inline int64_t hashData(bool value) const {
```

Comment:
I wonder if this is needed. I think this can be changed to VELOX_UNREACHABLE() or removed.